firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentRefContextualSerializer.java

```diff
@@ -0,0 +1,83 @@
+package com.google.firebase.firestore;
+
+import kotlin.Metadata;
+import kotlin.collections.CollectionsKt;
+import kotlin.jvm.internal.Reflection;
+import kotlin.reflect.KClass;
+import kotlinx.serialization.ContextualSerializer;
+import kotlinx.serialization.DeserializationStrategy;
+import kotlinx.serialization.KSerializer;
+import kotlinx.serialization.SerializationStrategy;
+import kotlinx.serialization.descriptors.SerialDescriptor;
+import kotlinx.serialization.encoding.Decoder;
+import kotlinx.serialization.encoding.Encoder;
+import kotlinx.serialization.modules.SerializersModule;
+import org.jetbrains.annotations.NotNull;
+
+@Metadata(
```

Comment:
I must have all this annotations (all these 4 fields), otherwise, I will have the NoELementException.